src/nupic/algorithms/Cells4.cpp

```diff
@@ -3176,3 +3177,37 @@ void Cells4::resetTimers()
   chooseCellsTimer.reset();
 #endif
 }
+
+bool Cells4::operator==(const Cells4& other) const
+{
+  // TODO: Add the rest of the fields!
+  if (this->nSegments() != other.nSegments() ||
+      this->nCells() != other.nCells() ||
+      this->nColumns() != other.nColumns() ||
+      this->nCellsPerCol() != other.nCellsPerCol() ||
+      this->getMinThreshold() != other.getMinThreshold() ||
+      this->getPermConnected() != other.getPermConnected() ||
+      this->getVerbosity() != other.getVerbosity() ||
+      this->getMaxAge() != other.getMaxAge() ||
+      this->getPamLength() != other.getPamLength() ||
+      this->getMaxInfBacktrack() != other.getMaxInfBacktrack() ||
+      this->getMaxLrnBacktrack() != other.getMaxLrnBacktrack() ||
+
+      this->getPamCounter() != other.getPamCounter() ||
+      this->getMaxSeqLength() != other.getMaxSeqLength() ||
+      this->getAvgLearnedSeqLength() != other.getAvgLearnedSeqLength() ||
+      this->getNLrnIterations() != other.getNLrnIterations() ||
+
+      this->getMaxSegmentsPerCell() != other.getMaxSegmentsPerCell() ||
+      this->getMaxSynapsesPerSegment() != other.getMaxSynapsesPerSegment() ||
+      this->getCheckSynapseConsistency() != other.getCheckSynapseConsistency())
```

Comment:
I think this is outside the scope of this test. I wanted to leave in this code, which was originally just debugging, because I thought a start was better than nothing. But I also think it was confusing as operator== so I moved it into the specific test with explicit notice that it isn't complete.